Update the supported platforms

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates the platforms supported by this Ansible role in addition to ensuring that the [cyhy-core] package is installed with `pip2`.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## ⚠ Note ##

This pull request is reliant on the merge of the following:

- cisagov/ansible-role-pip#50
- cisagov/ansible-role-python#46

## 💭 Motivation and context ##

We are in the process of updating the instances in [cisagov/cyhy_amis](https://github.com/cisagov/cyhy_amis) to use Debian Buster for instances running Python 2 code. With the above mentioned PRs for Python/pip installation we can expand the supported platforms.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been approved. -->

- [x] Revert dependencies to default branches.

[cyhy-core]: https://github.com/cisagov/cyhy-core

mcdonnnj committed:
Update the list of supported platforms

We are making Python 2/pip2 support non-platform specific so we can
expand the platforms we support. Comments that specify Debian Stretch
have been updated accordingly.

mcdonnnj committed:
Ensure that the cyhy-core package is installed with pip2

mcdonnnj committed:
Update the molecule prepare configuration

With changes to the cisagov/ansible-role-python,
cisagov/ansible-role-pip, and cisagov/ansible-role-remove-python2
projects we need to specify that we want to install Python 2/pip2 and
that we do not want to remove Python 2.

mcdonnnj committed:
Use branches for two roles in molecule testing

Use the branches for cisagov/ansible-role-python and
cisagov/ansible-role-pip that support installation of Python 2/pip2 as
a variable flag.

mcdonnnj committed:
Update molecule testing

In addition to testing that the cyhy-core package is installed we
should verify that all of the apt packages we need are installed.

jsf9k commented on molecule/default/python.yml:
```diff
@@ -4,6 +4,7 @@
-    - pip
-    - python
-    - remove_python2
+    - install_pip2: true
+      role: pip
+    - install_python2: true
+      role: python
```
Aren't role variables [normally handled like this](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#roles-keyword):
```suggestion
    - role: pip
      vars:
        install_pip2: true
    - role: python
      vars:
        install_python2: true
```

mcdonnnj replied:
Ah, sorry, I was just going with how it's done  in [cisagov/cyhy_amis](https://github.com/cisagov/cyhy_amis/blob/d368af9ab4c7e7e5481c325c04c431b2a50e2bbb/packer/ansible/playbook.yml#L42-L51) since it was fresh on my mind. I can change it if we'd prefer this way (and change what I linked accordingly).

jsf9k replied:
I prefer the way I suggested; even better might be to use `ansible.builtin.import_role` as I think that is the preferred way to do it now.  That said, I'm fine with whatever the group decides.

mcdonnnj replied:
If we go the `ansible.builtin.import_role` route I think that change should be made in [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and trickle down. I'll update to use your suggested syntax instead for this PR.

jsf9k reviewed (commented):
I have one question.

mcdonnnj committed:
Update variable passing for roles in the converge step

This method is clearer and better articulates what we are doing than
lumping it into a single map.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

dav3r reviewed (approved):
Looks groovy to me ⚡

jsf9k commented on molecule/default/molecule-no-systemd.yml:
```diff
@@ -24,8 +24,8 @@ platforms:
   - name: debian9
```
The comment directly after `platforms:` should be modified to agree with that was written in the role metadata.

mcdonnnj replied:
Please see [bf00fc9](https://github.com/cisagov/ansible-role-cyhy-core/pull/65/commits/bf00fc97d9269709f68d2237486306bc8eb2d202).

jsf9k commented on molecule/default/python.yml:
```diff
@@ -4,6 +4,9 @@
   become: yes
```
Update the play name directly above this line.

mcdonnnj replied:
Please see [1f58866](https://github.com/cisagov/ansible-role-cyhy-core/pull/65/commits/1f588661f2703dfe54c9892e458a232e78bf0a72).

jsf9k reviewed (changes requested):
I think the `dependencies` section of the role metadata should set the `install_pip2` variable for [cisagov/ansible-role-pip](https://github.com/cisagov/ansible-role-pip):
```yaml
  dependencies:
    - src: https://github.com/cisagov/ansible-role-geoip2
      name: geoip2
    - src: https://github.com/cisagov/ansible-role-pip
      name: pip
      vars:
        install_pip2: true
```

See [here](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#using-role-dependencies), for example.

I noticed this while working on cisagov/ansible-role-cyhy-reports#44, which pulls in this role as a dependency.

mcdonnnj committed:
Update a play name in the molecule configuration

The name did not correctly represent what was being done in the play.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Update a role dependency with a necessary variable

We need to ensure that cisagov/ansible-role-pip installs pip2 even if
only run as a dependency of this role.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Use consistent comments when explaining the same idea

Update a comment explaining platform use in the molecule configuration
to mirror what is said about the same topic in the role metadata.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

jsf9k reviewed (approved):
Now even more approved!

mcdonnnj committed:
Update molecule test for pip packages

The host.pip_package class has been deprecated so we need to switch to
using the host.pip class.

mcdonnnj committed:
Revert "Use branches for two roles in molecule testing"

This reverts commit 5822436ae272f3318e6c387b0c183448750bcf36.

The branches for these dependencies have been merged into their
respective develop branches.
